### DISCUSSION TITLE
Is it possible to write HTML in the editor and export it like that?

### DISCUSSION THREAD

**Discussion by buondevid** [facebook/lexical#5565] [Q&A]
I'm using the rich text plugin In Lexical React and when I export the text content it looks like Lexical escapes all the tags that I write in the editor.

For example, I want to allow writing in the editor `<b>Hello</b>` and have it exported exactly like that, but the tags get escaped.

You can see it happening in the official playground as well: 
<img width="1188" alt="image" src="https://github.com/facebook/lexical/assets/36935593/9bab69ae-e5d6-4430-bc22-58a8747bb021">

**Comment by 2wheeh** [comment]
How about this kind of workaround using `htmlConfig`(ref: https://github.com/facebook/lexical/releases/tag/v0.12.3) ?
Below logic is not complete as is. This would probably only work when `<b>foo</b>` is separated text node.
But you could extend this more (with regex maybe)?
take a look:

```ts
{
  export: new Map([
    [
      TextNode,
      (editor, node) => {
        const textNode = node as TextNode;
        const textContent = textNode.getTextContent();
        let element;

        if (textContent.startsWith('&lt;b&gt;') && textContent.endsWith('&lt;/b&gt;')) {
          const trimmed = /* trim <b> and </b> */ 
          element = document.createElement('b');
          element.append(document.createTextNode(trimmed));
        } else {
          element = textNode.exportDOM(editor).element;
        }

        return { element };
      },
    ],
  ]),
}
```

**Reply by buondevid** [reply]
yea thanks, this could work but it's a bit of a hassle... I was hoping for the existence of a configurable prop like `escapeHtml: false` eheh
